internal/datastore/spanner/spanner.go

```diff
@@ -115,11 +117,42 @@ func NewSpannerDatastore(database string, opts ...Option) (datastore.Datastore,
 	return ds, nil
 }
 
+type traceableRTX struct {
```

Comment:
but this is a read only transaction. I named it that way because I also had one wrapper for the readwrite transaction, but that I got rid eventually because wasn't useful. The struct only implements read only operations.